### PR TITLE
Altar item swap exploit fix

### PR DESCRIPTION
**Pull request by dniym**
Currently there is a bug with the dropped items used on an ancient altar, if a player uses a piston, or water to push the item off the pedestal (after starting the ritual), they can then push a less expensive item back on the pedestal and the ritual will complete as if the correct item was used.

Fix saves the items and locations when the ritual is initiated and if a change is detected the ritual is aborted.

Machine to duplicate exploit is here:  
http://prntscr.com/on2fk1

1)Load pedestals with actual materials required for ritual.
2)Load pedestal adjacent to actual pedestal with a dummy item ie dirt.
3)Start ritual
4) activate piston to push off valuable item, in this case Magical Lump
5) activate piston to push on dummy item.
6) ritual completes giving the desired output from the ritual and leaving behind the Magical Lump.

### CODE REVIEW

File: src/me/mrCookieSlime/Slimefun/AncientAltar/RitualAnimation.java [hunk 1]
```diff
@@ -82,7 +109,11 @@ private void checkPedestal(Block pedestal) {
 				e.printStackTrace();
 			}
 
+			
+			System.out.println("UnLocking item: " + item.getCustomName() + " / " + item.getItemStack().getType().name());
```
Comment (TheBusyBiscuit) [inline src/me/mrCookieSlime/Slimefun/AncientAltar/RitualAnimation.java]: Pretty sure that this debug message should be removed.

File: src/me/mrCookieSlime/Slimefun/Setup/SlimefunSetup.java [hunk 2]
```diff
@@ -3611,30 +3612,30 @@ public void registerDefaultRecipes() {
...
 				registerFuel(new MachineFuel(2, new ItemStack(Material.CACTUS)));
 				registerFuel(new MachineFuel(2, new ItemStack(Material.LILY_PAD)));
 				registerFuel(new MachineFuel(8, new ItemStack(Material.CHORUS_FRUIT)));
-
+				registerFuel(new MachineFuel(1, new ItemStack(Material.BAMBOO)));
+				registerFuel(new MachineFuel(1, new ItemStack(Material.KELP)));
+				registerFuel(new MachineFuel(2, new ItemStack(Material.DRIED_KELP)));
+				registerFuel(new MachineFuel(20, new ItemStack(Material.DRIED_KELP_BLOCK)));
+				registerFuel(new MachineFuel(1, new ItemStack(Material.SEAGRASS)));
+				registerFuel(new MachineFuel(2, new ItemStack(Material.SEA_PICKLE)));
```
Comment (TheBusyBiscuit) [inline src/me/mrCookieSlime/Slimefun/Setup/SlimefunSetup.java]: This change belongs to a different PR, so it should be excluded from this one.

File: src/me/mrCookieSlime/Slimefun/AncientAltar/RitualAnimation.java (hunk 1 above)
Comment (dniym) [reply src/me/mrCookieSlime/Slimefun/AncientAltar/RitualAnimation.java]: Doh yeah, I’ll fix that

File: src/me/mrCookieSlime/Slimefun/Setup/SlimefunSetup.java (hunk 2 above)
Comment (dniym) [reply src/me/mrCookieSlime/Slimefun/Setup/SlimefunSetup.java]: My bad, though for sure I added it as a new branch will fix shortly

**Pull request closed without merging**